domain/services/graphql/RegisterSchema.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+namespace EventEspresso\RecurringEvents\domain\services\graphql;
+
+use EventEspresso\core\domain\services\graphql\Utilities;
+use EventEspresso\core\services\graphql\fields\GraphQLFieldInterface;
+use EventEspresso\core\services\graphql\fields\GraphQLInputField;
+use GraphQLRelay\Relay;
+
+/**
+ * Class RegisterSchema
+ * Description
+ *
+ * @package EventEspresso\RecurringEvents\domain\services\graphql
+ * @author  Manzoor Wani
+ * @since   $VID:$
+ */
+class RegisterSchema
+{
+
+    /**
+     * @var Utilities
+     */
+    private $utilities;
+
+
+    /**
+     * RegisterSchema constructor.
+     *
+     * @param Utilities $utilities
+     */
+    public function __construct(Utilities $utilities)
+    {
+        $this->utilities = $utilities;
+    }
+
+
+    /**
+     * @return void
+     * @since $VID:$
+     */
+    public function addFilters()
+    {
+        add_filter(
+            'FHEE__EventEspresso_core_domain_services_graphql_types__datetime_fields',
+            [$this, 'registerCoreDatetimeFields']
+        );
+        add_filter(
+            'FHEE__EventEspresso_core_domain_services_graphql_data_mutations__datetime_args',
+            [$this, 'addDatetimeMutationArgs'],
+            10,
+            2
+        );
+        add_filter(
+            'FHEE__EventEspresso_core_domain_services_graphql_connections__datetime_args',
+            [$this, 'addDatetimeConnectionArgs']
+        );
+        add_filter(
+            'FHEE__EventEspresso_core_domain_services_graphql_connection_resolvers__datetime_where_params',
+            [$this, 'addDatetimeConnectionWhereParams'],
+            10,
+            3
+        );
+    }
+
+
+    /**
+     * @param GraphQLFieldInterface[] $fields
+     * @return GraphQLFieldInterface[]
+     * @since $VID:$
+     */
+    public function registerCoreDatetimeFields($fields)
+    {
+        $newFields = [
+            // add recurrence field to datetime mutation input
+            new GraphQLInputField(
+                'recurrence',
+                'String',
+                null,
+                sprintf(
+                    '%1$s %2$s',
+                    esc_html__('Globally unique ID of the recurrence related to the datetime.', 'event_espresso'),
+                    esc_html__('Ignored if empty.', 'event_espresso')
+                )
+            ),
+        ];
+
+        return array_merge($fields, $newFields);
+    }
+
+
+    /**
+     * @param array $args  The args that goto DB models.
+     * @param array $input Data coming from the GraphQL mutation query input.
+     * @return array
+     * @since $VID:$
+     */
+    public function addDatetimeMutationArgs($args, $input)
+    {
+        if (! empty($input['recurrence'])) {
+            $parts = Relay::fromGlobalId(sanitize_text_field($input['recurrence']));
+            $args['RCR_ID'] = (! empty($parts['id']) && is_int($parts['id'])) ? $parts['id'] : null;
```

Comment:
`is_int` check has caused issues in the past and this PR as well. `RCR_ID` is not saved to datetime table because of this.
```suggestion
            $args['RCR_ID'] = (! empty($parts['id'])) ? $parts['id'] : null;
```

Reply:
kk will change thnx